### PR TITLE
Fixed restoring the previous signal handler for example in python case (Windows only for now)

### LINKED ISSUE #3271: v2: Ctrl+C / SIGINT / KeyboardInterrupt handling not working
### Troubleshooting docs

- [X] My problem is not solved in the Troubleshooting docs

### Anaconda default channels

- [X] I do NOT use the Anaconda default channels (pkgs/* etc.)

### How did you install Mamba?

Other (please describe)

### Search tried in issue tracker

ctrl c handling

### Latest version of Mamba

- [X] My problem is not solved with the latest version

### Tried in Conda?

Not applicable

### Describe your issue

In the latest beta, `Ctrl+C` handling from Python doesn't work correctly. In v1, we added `Context.use_default_signal_handler(...)` as a workaround, but this code doesn't seem to work in v2 anymore.

This simple script reproduces the problem. Run it with `python` and try to `Ctrl+C` out of it:

```python
import time
from libmambapy import Context
# This line below doesn't have any effect;
# it can be commented out and the result is the same
Context.use_default_signal_handler(False)
print("Sleep for 10s... Try to Ctrl+C")
time.sleep(10)
```

It just gets stuck until the end of the 10s sleep:

```
$ python debug_ctrl_c.py 
Sleep for 10s... Try to Ctrl+C
^C^C^C^C^C^C^C^C
```

If you remove the libmambapy imports, the outcome is:

```
$ python debug_ctrl_c.py 
Sleep for 10s... Try to Ctrl+C
^CTraceback (most recent call last):
  File "/workspaces/conda-libmamba-solver/debug_ctrl_c.py", line 6, in <module>
    time.sleep(10)
KeyboardInterrupt
```

### mamba info / micromamba info

```shell
libmamba version : 2.0.0
          mamba version : 2.0.0
           curl version : libcurl/8.7.1 OpenSSL/3.2.1 zlib/1.2.13 zstd/1.5.5 libssh2/1.11.0 nghttp2/1.58.0
     libarchive version : libarchive 3.7.2 zlib/1.2.13 liblzma/5.2.6 bz2lib/1.0.8 liblz4/1.9.3 libzstd/1.5.5
       envs directories : /root/.local/share/mamba/envs
          package cache : /root/.local/share/mamba/pkgs
                          /root/.mamba/pkgs
            environment : /opt/conda (active)
           env location : /opt/conda
      user config files : /root/.mambarc
 populated config files : /opt/conda/.condarc
       virtual packages : __unix=0=0
                          __linux=5.15.49=0
                          __glibc=2.31=0
                          __archspec=1=aarch64
               channels : https://conda.anaconda.org/conda-forge/linux-aarch64
                          https://conda.anaconda.org/conda-forge/noarch
       base environment : /root/.local/share/mamba
               platform : linux-aarch64
```

### Logs

_No response_

### environment.yml

```yaml
N/A
```

### ~/.condarc

```yaml
N/A
```

### PR DESCRIPTION
**Pull request by Klaim**
Should fix #3271 partially.

### CODE REVIEW

**Comment by Klaim** [comment]
Trying to fix the linux case that still doesnt work.

**Comment by Klaim** [comment]
We'll try to get this merged to at least fix the Windows case, the non-working Linux case can be done in another PR

File: libmamba/src/core/thread_utils.cpp [hunk 1]
```diff
@@ -64,12 +67,15 @@ namespace mamba
         int signum = 0;
         // wait until a signal is delivered:
         sigwait(&sigset, &signum);
+        std::cerr << "default_signal_handler" << std::endl;
```
Comment (Hind-M) [inline libmamba/src/core/thread_utils.cpp:70]: Do we really want to keep these logs?

Comment (Klaim) [reply libmamba/src/core/thread_utils.cpp:70]: Woops not at all

File: libmamba/src/core/thread_utils.cpp [hunk 2]
```diff
@@ -5,6 +5,7 @@
 // The full license is in the file LICENSE, distributed with this software.
 
 #include <atomic>
+#include <iostream>
```
[the remark this answers is not in the document]
Comment (Klaim) [reply libmamba/src/core/thread_utils.cpp:8]: 🥲

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**